src/Repository/Storable.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+namespace Friendica\Repository;
+
+use Friendica\Collection;
+use Friendica\Model;
+use Friendica\Repository;
+use Friendica\Database\Database;
+use Friendica\Database\DBA;
+use Friendica\Network\HTTPException;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Storable repositories are linked to a main table with an id primary key and will instanciate Models and Collections
+ * related to that table.
+ */
+abstract class Storable extends Repository
```

Comment:
A `Repository` is per se a class for interacting with a storable backend, so what you mean with `Storable` **is** a `Repository`, the other classes are factories.

Reply:
Ok for the names.